jdaviz/configs/imviz/plugins/footprints/footprints.py

```diff
@@ -23,6 +24,54 @@
 __all__ = ['Footprints']
 
 
+# These functions help us determine which polygon overlay is spatially closest
+# to a user's click on the viewer. The math is based on projecting a point
+# onto line segments and computing the nearest one across all polygons.
+#
+# This is useful when multiple regions are displayed and we want to know
+# which one the user meant to interact with.
+def closest_point_on_segment(px, py, x1, y1, x2, y2):
+    """Find the closest point on a line segment to a given point."""

```

Comment:
nitpick (readability/maintainability): It would be helpful to define in the doc string what the various inputs mean
```suggestion
    """
    Find the closest point on a line segment to a given point.
    
    Parameters
    -----------
        px: int
            x coordinate of the point we are checking
        py: int
            y coordinate of the point we are checking
        x1: int
            x coordinate of the first point of the line segment
        y1: int
            y coordinate of the first point of the line segment
        x2: int
            x coordinate of the second point of the line segment
        y2: int
            y coordinate of the second point of the line segment
        
    Returns
    --------
        closest_x, clostest_y: tuple[int]
            the closest x and y coordinate to the point on the line segment
    """
```